x-pack/docs/en/rest-api/security/saml-invalidate-api.asciidoc

```diff
@@ -0,0 +1,87 @@
+[role="xpack"]
+[[security-api-saml-invalidate]]
+=== SAML invalidate API
+
+Submits a SAML LogoutRequest message to {es} for consumption. 
+
+[[security-api-saml-invalidate-request]]
+==== {api-request-title}
+
+`POST /_security/saml/invalidate`
+
+[[security-api-saml-invalidate-desc]]
+==== {api-description-title}
+
+The logout request comes from the SAML IdP during an IdP initiated Single Logout.
+After successful validation of the request, {es} invalidates the access token
+and refresh token that corresponds to that specific SAML principal and provides
+a URL that contains a SAML LogoutResponse message, so that the user can be
+redirected back to their IdP.
+
+{es} exposes all the necessary SAML related functionality via the SAML APIs.
+These APIs are used internally by {kib} in order to provide SAML based
+authentication, but can also be used by other custom web applications or other
+clients. See also <<security-api-saml-authenticate,SAML authenticate API>>,
+<<security-api-saml-prepare-authentication,SAML prepare authentication API>>,
+and <<security-api-saml-logout,SAML logout API>>.
+
+[[security-api-saml-invalidate-request-body]]
+==== {api-request-body-title}
+
+`acs`::
+  (Optional, string) The Assertion Consumer Service URL that matches the one of the SAML
+  realm in {es} that should be used. You must specify either this parameter or the `realm` parameter.
+
+`queryString`:: 
+  (Required, string) The query part of the URL that the user was redirected to by the SAML
+  IdP to initiate the Single Logout. This query should include a single
+  parameter named `SAMLRequest` that contains a SAML logout request that is
+  deflated and Base64 encoded. If the SAML IdP has signed the logout request, 
+  the URL should include two extra parameters named `SigAlg` and `Signature`
+  that contain the algorithm used for the signature and the signature value
+  itself.
```

Comment:
Per the SAML HTTP-Redirect spec, the Signature must be the last parameter in the query string (we strip it off in order to get the original text on which to perform signature verification), and in order to get the right signature the query string needs to be a _verbatim_ copy of what was received from the browser, it cannot be reconstructed from a parsed URL.

I think it's worth saying something along those lines here.

```suggestion
  itself. In order for {es} to be able to verify the IdP's signature, the value
  of the queryString field must be an exact match to the string provided
  by the browser. The client application must not attempt to parse or 
  process the string in any way.
```